flink-clients/src/main/java/org/apache/flink/client/cli/GenericCLI.java

```diff
@@ -105,44 +87,33 @@ public void addRunOptions(Options baseOptions) {
 	public void addGeneralOptions(Options baseOptions) {
 		baseOptions.addOption(executorOption);
 		baseOptions.addOption(targetOption);
-		baseOptions.addOption(dynamicProperties);
+		baseOptions.addOption(DynamicPropertiesUtil.DYNAMIC_PROPERTIES);
 	}
 
 	@Override
-	public Configuration applyCommandLineOptionsToConfiguration(final CommandLine commandLine) {
-		final Configuration effectiveConfiguration = new Configuration(baseConfiguration);
+	public Configuration toConfiguration(final CommandLine commandLine) {
+		final Configuration resultConfiguration = new Configuration();
+
+		if (configuration.getOptional(DeploymentOptions.TARGET).isPresent()) {
```

Comment:
Can you explain a bit why we do not remove this `if() {}` block? If it is in the base configuration, we will add it later, right?

Reply:
I posted this answer earlier (https://github.com/apache/flink/pull/13554#discussion_r507626412):

> I see your point. For now we still use the Configuration in isActive(). We could think about not keeping the Configuration in the CLI but instead pass it as a parameter there as well. Also, I'm forwarding the option for now to make the returned Configuration "self contained", i.e. to have everything in there that the executor would expect.
> 
> What do you think?

Reply:
I know, but I do not really get it. I am not suggesting to remove the configuration from the class. I am just suggesting to not put the `TARGET` in the `resultConfiguration` but simply put whatever the user has put in the command line. I think that this is more consistent with the contract and also how the other custom command lines work.

Reply:
Ahh yes, you're right. I wasn't thinking straight.